test/uk/gov/hmrc/residencenilratebandcalculator/connectors/RnrbConnectorSpec.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.residencenilratebandcalculator.connectors
+
+import org.mockito.ArgumentCaptor
+import org.mockito.Matchers._
+import org.mockito.Mockito.{verify, when}
+import org.scalatest.mock.MockitoSugar
+import play.api.http.Status
+import play.api.libs.json._
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpReads, HttpResponse}
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+import uk.gov.hmrc.residencenilratebandcalculator.WSHttp
+import uk.gov.hmrc.residencenilratebandcalculator.models.CalculationResult
+
+import scala.concurrent.Future
+
+class RnrbConnectorSpec extends UnitSpec with WithFakeApplication with MockitoSugar {
+  "RNRB Connector" must {
+
+    "call the Microservice with the given JSON" in {
+      implicit val headerCarrierNapper = ArgumentCaptor.forClass(classOf[HeaderCarrier])
+      implicit val httpReadsNapper = ArgumentCaptor.forClass(classOf[HttpReads[Any]])
+      implicit val jsonWritesNapper = ArgumentCaptor.forClass(classOf[Writes[Any]])
+      val urlNapper = ArgumentCaptor.forClass(classOf[String])
+      val bodyNapper = ArgumentCaptor.forClass(classOf[JsValue])
+      val headersNapper = ArgumentCaptor.forClass(classOf[Seq[(String, String)]])
+      val httpMock = mock[WSHttp]
+      when(httpMock.POST(anyString, any[JsValue], any[Seq[(String, String)]])(any[Writes[Any]], any[HttpReads[Any]],
+        any[HeaderCarrier])) thenReturn Future.successful(HttpResponse(Status.OK, Some(Json.parse("{}"))))
```

Comment:
Could this mock setup come out into a function / method - it's repeated in every test